### PR TITLE
INDEV-10502 Github actions + Adopt backwards compatibility checker

### PR DESCRIPTION
**Pull request by CMalvika**
Integrate https://github.com/Roave/BackwardCompatibilityCheck to make versioning determination more trivial.
Also, configure Github actions.

BC Checker complains will go away after increasing the minor version! Here's the result from the local check, all good!
Screenshot 2022-04-07 at 1 51 27 PM

### CODE REVIEW

**Review by lstrojny — APPROVED** [review]

File: tests/InterNations/Tests/Integration/BundleIntegrationTest.php [hunk 1]
```diff
@@ -1,11 +1,10 @@
 <?php
```
Comment (CMalvika) [inline tests/InterNations/Tests/Integration/BundleIntegrationTest.php]: @lstrojny how bad an idea is it that I only moved BundleIntegrationTest to ./tests?
- With that, the test is running as expected
- Also BC checker is fine (Still need to increase minor version)

Comment (lstrojny) [reply tests/InterNations/Tests/Integration/BundleIntegrationTest.php]: Why not move all the tests and have them in one folder?

Comment (CMalvika) [reply tests/InterNations/Tests/Integration/BundleIntegrationTest.php]: I tried to do that first! But i was gettig these kind of errors for all the files which are under /app and /fixture.

14) InterNations\Bundle\TypeJailBundle\Tests\Integration\BundleIntegrationTest::testRenderAggregateTemplate
Error: Class 'InterNations\Bundle\TypeJailBundle\Tests\Integration\app\AppKernel' not found

/Users/malvikachauhan/Projects/type-jail-bundle/tests/InterNations/tests/Integration/BundleIntegrationTest.php:99
/Users/malvikachauhan/Projects/type-jail-bundle/tests/InterNations/tests/Integration/BundleIntegrationTest.php:86
phpvfscomposer:///Users/malvikachauhan/Projects/type-jail-bundle/vendor/phpunit/phpunit/phpunit:97

Comment (lstrojny) [reply tests/InterNations/Tests/Integration/BundleIntegrationTest.php]: Did you configure a dev auto loader?

Comment (CMalvika) [reply tests/InterNations/Tests/Integration/BundleIntegrationTest.php]: I configured testsuite path. Where else i need configuration?
```
<testsuites>
        <testsuite name="internations/type-jail-bundle">
            <directory>./tests</directory>
        </testsuite>
</testsuites>
```

Comment (lstrojny) [reply tests/InterNations/Tests/Integration/BundleIntegrationTest.php]: Check this out https://getcomposer.org/doc/04-schema.md#autoload-dev

Comment (CMalvika) [reply tests/InterNations/Tests/Integration/BundleIntegrationTest.php]: I will do, thank you!

Comment (CMalvika) [reply tests/InterNations/Tests/Integration/BundleIntegrationTest.php]: Done! And it's working now. Thank you very much. I guess now i am ready to merge all the 4 PRs, and bump the minor version for this project.

Comment (lstrojny) [reply tests/InterNations/Tests/Integration/BundleIntegrationTest.php]: Congrats and yes, looks good

**Pull request merged**